community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyStore.java

```diff
@@ -326,20 +330,59 @@ else if ( block.getType() == PropertyType.ARRAY )
     @Override
     public PropertyRecord getRecord( long id )
     {
-        try ( PageCursor cursor = storeFile.io( pageIdForRecord( id ), PF_SHARED_LOCK ) )
+        return getRecord( new PropertyRecord( id ) );
+    }
+
+    public PropertyRecord getRecord( PropertyRecord record )
+    {
+        try ( PageCursor cursor = storeFile.io( pageIdForRecord( record.getId() ), PF_SHARED_LOCK ) )
         {
-            PropertyRecord record = null;
+            PropertyRecord tmpRecord = null;
             if ( cursor.next() )
             {
                 do
                 {
-                    record = getRecord( id, cursor );
+                    tmpRecord = getRecord( cursor, record );
                 } while ( cursor.shouldRetry() );
             }
 
-            if ( record == null || !record.inUse() )
+            if ( tmpRecord == null || !tmpRecord.inUse() )
             {
-                throw new InvalidRecordException( "PropertyRecord[" + id + "] not in use" );
+                throw new InvalidRecordException( "PropertyRecord[" + record.getId() + "] not in use" );
+            }
+
+            return record;
+        }
+        catch ( IOException e )
+        {
+            throw new UnderlyingStorageException( e );
+        }
+    }
+
+    public PageCursor newPageCursor(int mode) throws IOException
+    {
+        return storeFile.io(0, mode);
+    }
+
+    public PropertyRecord getRecord( PropertyRecord record, PageCursor cursor )
```

Comment:
Can we avoid having `getRecord(PropertyRecord,PageCursor)` and `getRecord(PageCursor, PropertyRecord)` definied here? Perhaps renaming them to something else?